ann_benchmarks/distance.py

```diff
@@ -1,20 +1,38 @@
 from __future__ import absolute_import
 from scipy.spatial.distance import pdist as scipy_pdist
-
+import numpy as np
 
 def pdist(a, b, metric):
     return scipy_pdist([a, b], metric=metric)[0]
 
-# Need own implementation of jaccard because numpy's
+# Need own implementation of jaccard because scipy's
 # implementation is different
 
-
 def jaccard(a, b):
     if len(a) == 0 or len(b) == 0:
-        return 0
-    intersect = len(a & b)
+         return 0
+    intersect = len(set(a) & set(b))
     return intersect / (float)(len(a) + len(b) - intersect)
 
+def transform_dense_to_sparse(X):
+    """Converts the n * m dataset into a sparse format
+    that only holds the non-zero entries (Jaccard)."""
+    # get list of indices of non-zero elements
+    indices = np.transpose(np.where(X))
```

Comment:
not super important but you could probably use `itertools.groupby` here:
```python
indices = np.transpose(np.where(X))
keys = []
for ijs in itertools.groupby(indices,  lambda ij: ij[0]):
    keys.append([j for i, j in ijs])
return keys
```
haven't tested it though

Reply:
Thanks, worked almost out-of-the-box. :-)